src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/service/PrisonOffenderEventListener.kt

```diff
@@ -36,10 +40,18 @@ class PrisonOffenderEventListener(
         )
       }
       PRISONER_BOOKING_MOVED_EVENT_TYPE -> {
-        val mergeEvent = mapper.readValue(message, HMPPSMergeBookingMovedEvent::class.java)
-        reportService.replacePrisonerNumber(
-          removedPrisonerNumber = mergeEvent.additionalInformation.movedFromNomsNumber,
-          prisonerNumber = mergeEvent.additionalInformation.movedToNomsNumber,
+        val moveEvent = mapper.readValue(message, HMPPSMergeBookingMovedEvent::class.java)
+        val additionalInformation = moveEvent.additionalInformation
+        reportService.replacePrisonerNumberInDateRange(
+          removedPrisonerNumber = additionalInformation.movedFromNomsNumber,
+          prisonerNumber = additionalInformation.movedToNomsNumber,
+          // use booking date range if and only if `bookingStartDateTime` is specified
+          since = additionalInformation.bookingStartDateTime,
+          until = if (additionalInformation.bookingStartDateTime != null) {
```

Comment:
This is probably a stupid question but since and until are set from the same `additionalInformation.bookingStartDateTime` field?

Oh, I see if since is `bookingStartDateTime` is set, the event `occurredAt` is used as `until` 👌